process-dataset/steps/upload-feature-defs.js

```diff
@@ -8,7 +10,15 @@ const uploadFeatureDefs = async (firebaseHandler, featureDefs) => {
     console.log("uploading feature defs...")
     for (let index = 0; index < featureDefs.length; index++) {
         const feature = featureDefs[index];
-
+        if (feature.discrete) {
+            // check if the keys in the options are unique, if they exist
+            const noOptions = map(feature.options).length;
+            const noUniqueKeys = filter(uniq(map(feature.options, "key"))).length;
```

Comment:
I'm a little confused here.. Is `noOptions` and `noUniqueKeys` supposed to stand for "number of options" and "number of unique keys"? Is `features.options` not the same thing as `map(feature.options)`?

Reply:
sorry yeah I can see how that would be confusing. I'll change them to `num`

Features.options is an object with a bunch of 
```
number: {
     name: string;
     key?: string;
```
Objects. I mapped over them to give me an array since objects don't have lengths. I can make this clearer though

Reply:
i updated it, let me know if that's clearer

Reply:
Okay now I get it haha. Thanks! Could you also change `noUniqueKeys` to `numUniqueKeys` though? The `no` really throws me off, it makes me think it's a boolean indicating there are no unique keys

Reply:
yeah sorry, I meant to do that too